File: src-common/interfaces/IProviderService.ts
import { ProviderModel, ProviderType } from '../models/provider.model';
import { DirectoryModel } from '../models/directory.model';
import { BaseService } from '../base.service';

export interface IProviderService extends BaseService {
    readonly Type: ProviderType;
    readonly IsRootDrive: boolean;
    GetProvider(): Promise<ProviderModel>;
    GetDirectory(path: string): Promise<DirectoryModel>;
}

File: Jaya.Ui/Services/Providers/IProviderService.cs
using Jaya.Ui.Models;

namespace Jaya.Ui.Services.Providers
{
    public interface IProviderService
    {
        bool IsRootDrive { get; }

        string Name { get;  }

        DirectoryModel GetDirectory(ProviderModel provider, string path = null);

        ProviderModel GetDefaultProvider();
    }
}
